addons/mail/models/mail_mail.py

```diff
@@ -40,9 +41,16 @@ def default_get(self, fields):
 
     # content
     mail_message_id = fields.Many2one('mail.message', 'Message', required=True, ondelete='cascade', index=True, auto_join=True)
+
+    # Mail Message fields
+    message_int = fields.Integer(compute='_compute_message_int', compute_sudo=True)
+
     body_html = fields.Text('Rich-text Contents', help="Rich-text/HTML message")
     references = fields.Text('References', help='Message references, such as identifiers of previous messages', readonly=1)
     headers = fields.Text('Headers', copy=False)
+    restricted_attachment_count = fields.Integer('Restricted attachments', compute='_compute_restricted_attachments')
+    unrestricted_attachment_ids = fields.Many2many('ir.attachment', string='Unrestricted Attachments',
```

Comment:
Would completely love having an unit test for that (especially with the inverse as this is nice, but definitively prone to be broken someday :D ) .